### PR TITLE
alpha-resize and sharpen rgb

### PR DESCRIPTION
**Pull request by alon-ne**
Use Alpha Resize for downscaling images with alpha channel and sharpen in RGB mode

### CODE REVIEW

File: vips/image.go [hunk 1]
```diff
@@ -654,20 +654,24 @@ func (r *ImageRef) Invert() error {
...
 	if err != nil {
 		return err
 	}
+	r.setImage(out)
 
-	out, err := vipsResize(r.image, scale, kernel)
+	return nil
+}
+
+func (r *ImageRef) AlphaResize(scale float64) error {
```
Comment (Elad-Laufer) [inline vips/image.go:667]: Please don't expose the alpha details to the outside, in the Resize function check for alpha and call the proper **private** function accordingly

Comment (alon-ne) [reply vips/image.go:667]: When doing the resize on images with alpha the algorithm parameter is ignored, I don't want govips interface to lie

Comment (Elad-Laufer) [reply vips/image.go:667]: I rather lie and add the algo support later on (if needed...), getting rid of public function is pain.

The only reason the the algo argument is mandatory and not optional is that Go does not support overloading function

Comment (alon-ne) [reply vips/image.go:667]: ok

Comment (alon-ne) [reply vips/image.go]: done

File: vips/image.go [hunk 2]
```diff
@@ -654,20 +655,22 @@ func (r *ImageRef) Invert() error {
...
-	err := r.PremultiplyAlpha()
-	if err != nil {
-		return err
+	var out *C.VipsImage
+	var err error
+
+	if scale < 1 && r.HasAlpha() {
+		out, err = vipsAlphaResize(r.image, scale)
+	} else {
+		out, err = vipsResize(r.image, scale, kernel)
```
Comment (Elad-Laufer) [inline vips/image.go:665]: I think we might be missing the PremultiplyAlpha() when scale > 1, but I might be wrong...

Comment (alon-ne) [reply vips/image.go:665]: aaah you're right, I'll do that. Good thing we have code review.
Just to make sure, this scenario doesn't actually exist in image manipulator right?

Comment (Elad-Laufer) [reply vips/image.go]: scale > 1? it exist, we don't prohibit upscaling and in some scenarios do them intently (as a fallback of superscale)

Comment (alon-ne) [reply vips/image.go]: So we might still get the alpha artifacts when upscaling

Comment (Elad-Laufer) [reply vips/image.go]: Upscaling alpha containing images is allowed

**Pull request merged**